animation_nodes/nodes/mesh/set_vertex_weights.py

```diff
@@ -0,0 +1,145 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... base_types import AnimationNode, VectorizedSocket
+
+modeItems = [
+    ("ALL", "All", "Set weight of every vertex", "NONE", 0),
+    ("INDEX", "Index", "Set weight of a specific vertex", "NONE", 1)
+]
+
+groupIdentifierTypeItems = [
+    ("INDEX", "Index", "Get vertex group based on the index", "NONE", 0),
+    ("NAME", "Name", "Get vertex group based on the name", "NONE", 1)
+]
+
+noMeshMessage = "no mesh object"
+editMeshMessage = "object is not in object mode"
+
+class VertexWeights(bpy.types.Node, AnimationNode):
+    bl_idname = "an_SetVertexWeights"
+    bl_label = "Set Vertex Weights"
+    errorHandlingType = "EXCEPTION"
+
+    mode: EnumProperty(name = "Mode", default = "ALL",
+        items = modeItems, update = AnimationNode.refresh)
+
+    groupIdentifierType: EnumProperty(name = "Group Identifier Type", default = "INDEX",
+        items = groupIdentifierTypeItems, update = AnimationNode.refresh)
+
+    useIndexList: VectorizedSocket.newProperty()
+    useFloatList: VectorizedSocket.newProperty()
```

Comment:
Use a more meaningful name, like `useWeightsList`.

Reply:
I have renamed the _useFloatList_ to _useWeightsList_.